amiqus/models/record.py

```
@@ -34,22 +34,35 @@ def create_record(self, client: Client, raw: dict) -> Record:
             raw=raw,
         )
 
-        # Now handle the checks/steps after record is saved
+        # Create all related objects
+        self._create_steps_from_response(record, raw)
+
+        return record
+
+    def _create_steps_from_response(self, record: Record, raw: dict) -> None:
+        """Create Steps and their related Check/Form objects from API response."""
         from .check import Check
+        from .form import Form
+        from .step import Step
 
-        for step in raw.get("steps", []):
-            if step.get("check"):
-                # Use get_or_create but with the saved record
-                Check.objects.get_or_create(
-                    amiqus_id=step["check"],
+        for step_data in raw.get("steps", []):
+            if "check" in step_data.get("type"):
+                check = Check.objects.get_or_create(
```

Why are we doing `get_or_create` instead of `create`? Given this is a new record, how can the check exist already?

---

➕ I'd definitely avoid `get_or_create` _if_ this particular part should always be creating, it tends to cause more issues than it's worth if there's not a good reason for it existing.

---

Ah yeah. The previous code was wrong to do so if at the point in time the line is called its always meant to be doing creation. That way we hear about errors when objects already exist as we should rather than them being silent errors. And we should hear about them because if that has happened it busts our understanding of how we think its working 👍

---

To add to that, my understanding is that once a Record is created with associated checks, it cannot change, i.e. if another check of the same type is needed, a whole new Record would need to be created, with a new check. So another +1 for only using `create`.